document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -608,6 +641,35 @@ public void drop() {
     }
   }
 
+  private Map<String, String> getDocIdToTenantIdMap(BulkArrayValueUpdateRequest request) {
+    return request.getKeys().stream()
+        .collect(
+            Collectors.toMap(
+                key -> key.toString().split(":")[1], key -> key.toString().split(":")[0]));
```

Comment:
`Key` is an interface and `SingleValueKey` is just one implementation of the interface. Here, we seem to be assuming that all `Key` are `SingleValueKey`, which doesn't seem to be a fair assumption (since an interface is meant to be overridden).

Reply:
I see this has been moved from a different place. But, I'm still not convinced how are we assuming that `key.toString().split(":")[1]` does not throw `ArrayIndexOutOfBoundsException` in all the cases.

Reply:
Let me refactor this code. Just re-arranged as part of this PR.